components/org.wso2.carbon.crypto.provider/src/main/java/org/wso2/carbon/crypto/provider/SymmetricKeyInternalCryptoProvider.java

```diff
@@ -131,17 +153,110 @@ public SymmetricKeyInternalCryptoProvider(String secretKey) {
     public byte[] encrypt(byte[] cleartext, String algorithm, String javaSecurityAPIProvider,
                           boolean returnSelfContainedCipherText) throws CryptoException {
 
-        byte[] encryptedKey = encrypt(cleartext, algorithm, javaSecurityAPIProvider);
-        //When symmetric encryption algorithm with padding is introduced, we can create self contained cipher texts.
-        //Then we can utilize the returnSelfContainedCipherText parameter and add the necessary logic to provide a
-        // self contained cipher text. As of now, no cipher meta data is included in the cipher text. It contains
-        // only the cipher.
-        return encryptedKey;
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("Encrypting data with symmetric key encryption with algorithm: '%s'.", algorithm));
+        }
+        if (AES_GCM_SYMMETRIC_CRYPTO_ALGORITHM.equals(algorithm)) {
+            return encryptWithGCMMode(cleartext, javaSecurityAPIProvider);
+        }
```

Comment:
This should be extensible such that based on algorithm encryption happens so that we can easily introduce algorithms later.
Let's create a git issue for this improvement and track

Reply:
please find the git issue https://github.com/wso2/product-is/issues/8019